lib/rubocop/cop/github/rails_link_href.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "rubocop"
+
+module RuboCop
+  module Cop
+    module GitHub
+      module Accessibility
+        class LinkHref < Base
+          MSG = "Links should go somewhere, you probably want to use a `<button>` instead.".freeze
+
+          def on_send(node)
+            receiver, method_name, *args = *node
+
+            if receiver.nil? && method_name == :link_to
+              if args.first.type == :str && args.first.children.first == "#"
```

Comment:
Does the test fail?

Maybe we should also add a test like this?

```
  def test_link_href_hash_offense
    offenses = erb_investigate cop, <<-ERB, "app/views/products/index.html.erb"
      <%= link_to 'Go to GitHub', '#' %>
    ERB

    assert_equal 1, offenses.count
    assert_equal "Links should go somewhere, you probably want to use a `<button>` instead.", offenses[0].message
  end
```

Seems like our code should pass on all three of the tests including this one?

Reply:
ooh yeah this logic does seem off. nice catch!

it is checking if the label is `#` rather than the href. I was looking through [link_to docs](https://apidock.com/rails/ActionView/Helpers/UrlHelper/link_to) and it looks like there's various formats supported like:

```
link_to "GitHub", "#"

link_to nil, "http://github.com" 

link_to @profile do 
   Mona's profile
end 
```

We can iterate on this linter in the future but maybe for now we just check if the second argument exists, check if it's a string, check if it's `#` and use the test case @owenniblock proposed?

Reply:
as a future follow-up, we could add support for block syntax like: 

```
link_to '#' do 
   Mona's profile
end 
```